source/game/field/obj/ObjectKinoko.cc

```diff
@@ -0,0 +1,116 @@
+#include "game/field/obj/ObjectKinoko.hh"
+
+namespace Field {
+
+ObjectKinokoNm::ObjectKinokoNm(const System::MapdataGeoObj &params) : ObjectKCL(params) {
+    m_type = static_cast<KinokoType>(params.setting(0));
+}
+
+/// @addr{0x80827A9C}
+ObjectKinokoNm::~ObjectKinokoNm() = default;
+
+/// @addr{0x8080761C}
+ObjectKinoko::ObjectKinoko(const System::MapdataGeoObj &params)
+    : ObjectKCL(params), m_objPos(m_pos), m_objRot(m_rot) {
+    m_type = static_cast<KinokoType>(params.setting(0));
+
+    _12e = 0;
+    _144 = 6;
+    _130 = 40;
+    _132 = 10;
+    _12c = 1 % 40;
+    _134 = 0.1f;
+    m_period = 6.0f * F_PI / 40.0f;
+    _13c = 0.0008f;
+    _146 = 0;
```

Comment:
```suggestion
```
This likely isn't a part of the `ObjectKinoko` class, and is instead part of a derived class.

Reply:
This is confirmed to be in the base class, disregard. We can label this as `m_currentFrame`.